test/end-to-end/offer.js

```diff
@@ -603,4 +604,71 @@ describe("Offer endpoint tests", () => {
             });
         });
     });
+
+    describe("GET /offers/offerId", () => {
+
+        beforeAll(async () => {
+            await Offer.deleteMany({});
+        });
+
+        describe("Id Validation", () => {
+            test("Invalid ID", async () => {
+                const res = await request()
+                    .get("/offers/new/123");
+
+                expect(res.status).toBe(HTTPStatus.NOT_FOUND);
+                expect(res.body).toHaveProperty("reason", ValidationReasons.OBJECT_ID);
+            });
+            test("Offer not found", async () => {
+                const id = "5facf0cdb8bc30016ee58952";
```

Comment:
Wrap this in a Types.ObjectId from mongoose, to make sure that this string is always a valid mongo objected, otherwise it will throw something and the test fails. 

Then to be used below you might need to use id.toString(), but I'm not sure so try without it first but keep it in mind